Task3_Yoav_Uzan/include/Board.h

```diff
@@ -21,14 +21,12 @@ class Board
 	bool checkIllegelMoveRes21(Loction loc, Loction sourceLoc);// check if the position is legal for response 21
 	int checkChess(Loction loc, Loction sourceLoc);// check if the position is legal for response 31
 	bool checkIfKingInDanger(Loction KingLoc, Color kingColor, vector<vector<Peice*>> tempBoard);// check if the king is in danger

```

Comment:
the logic behind all these functions is not correct, you don't need all these function and to check each type of pieces with the king